pystac/item.py

```diff
@@ -431,25 +434,22 @@ def from_dict(
             raise pystac.STACTypeError(
                 f"{d} does not represent a {cls.__name__} instance"
             )
+        # These fields must exist on the dict for `matches_object_type` to pass
+        d.pop("type")
+        d.pop("stac_version")
 
-        if preserve_dict:
-            d = deepcopy(d)
-
-        id = d.pop("id")
-        geometry = d.pop("geometry")
-        properties = d.pop("properties")
+        id = d.pop("id", None)
+        geometry = d.pop("geometry", None)
+        properties = d.pop("properties", {})
         bbox = d.pop("bbox", None)
-        stac_extensions = d.get("stac_extensions")
+        stac_extensions = d.get("stac_extensions", None)
         collection_id = d.pop("collection", None)
 
         datetime = properties.get("datetime")
         if datetime is not None:
             datetime = str_to_datetime(datetime)
-        links = d.pop("links")
-        assets = d.pop("assets")
-
-        d.pop("type")
-        d.pop("stac_version")
+        links = d.pop("links", [])
+        assets = d.pop("assets", {})
 
```

Comment:
I think a more clear approach here would be to use `in` or `get` for accessing all of the fields that are expected, and then explicitly removing all of the ones that we don't want to be in extra_fields in one go, like:

```
for k in ["type", "stac_version", ....]:
  d.pop(k, None)
```

Reason is, I think it's unclear to someone reading this that the pops are both checking that the field is defined **and** modifying the dict to be set in the extra_fields. Separating the operations to validate the fields and modify the fields is less likely to cause bugs in the future if someone adds other checks that use pop or get, but wasn't thinking about how that effects the extra_fields.

Reply:
I totally agree that it is not clear from reading this the first time what is going on. I also didn't originally understand that the data is being paired down and then fed to extra_fields. I actually think this could do with more of a reorg not that I think about it.

Reply:
Very nice.